### PR TITLE
Fix get block

### PR DESCRIPTION
**Pull request by crypto-facs**
Closes: #778 

## Description
We were seeing an issue where some txs where not being indexed on our indexer without any noticeable errors. @yihuang noticed that tx Indexer runs asynchronously with block commit. This leads to unexepected behaviour since`eth_getBlockByNumber` currently cross checks valid txs based on the tx Indexer. `resultBlock` is on sync with block commit so it provides a better approach to remove failed transactions from the block response.

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/ethereum/backend/backend.go [hunk 1]
```diff
@@ -374,10 +381,9 @@ func (e *EVMBackend) EthBlockFromTendermint(
 
 			tx := ethMsg.AsTransaction()
 
-			// check tx exists on EVM and it has the correct block height
-			ethTx, err := e.GetTxByEthHash(tx.Hash())
-			if err != nil || ethTx.Height != block.Height {
-				e.logger.Debug("failed to query eth tx", "hash", tx.Hash().Hex())
+			// check tx exists on EVM by cross checking with blockResults
+			if txResults[i].Code != 0 {
```
Comment (leejw51crypto) [inline rpc/ethereum/backend/backend.go:385]: if Code is not 0, it's EVM tx?
maybe add some comment about it

**Review by leejw51crypto — APPROVED** [review]
lgtm

File: rpc/ethereum/backend/backend.go (hunk 1 above)
Comment (yihuang) [reply rpc/ethereum/backend/backend.go:385]: if `Code != 0`, it's a failed transaction.

**Review by fedekunze — COMMENTED** [review]
LGTM. Can you add a  bugfix changelog entry and an integration test on `tests/e2e/integration_test.go` to check for this?

**Review by fedekunze — APPROVED** [review]

**Pull request merged**